processing/proc_MARA.m

```diff
@@ -0,0 +1,299 @@
+    p2.y = sp_ica.x(find(sp_ica.t == p2.x,1),ic);
+
+    %third point: local minimum in the band 5-13 Hz
+    i = find(sp_ica.t == 5,1); 
+    stopp = find(sp_ica.t == 13,1); 
+    while sp_ica.x(i,ic) > sp_ica.x(i+1,ic) && i <= stopp
+        i = i +1; 
+    end
+    p3.x = sp_ica.t(i);
+    p3.y = sp_ica.x(i,ic); 
+
+    %fourth point: min - 1 in band 5-13 Hz
+    p4.x = p3.x - 1;
+    p4.y = sp_ica.x(find(sp_ica.t == p4.x,1),ic);
+        
+    %fifth point: local minimum in the band 33-39 Hz 
+    p5.y = min(sp_ica.x(find(sp_ica.t == 33,1):find(sp_ica.t == 39,1),ic));
+    p5.x = sp_ica.t(find(sp_ica.x(:,ic) == p5.y,1));
+
+    %sixth point: min + 1 in band 33-39 Hz
+    p6.x = p5.x + 1;
+    p6.y = sp_ica.x(find(sp_ica.t == p6.x,1),ic);
+    
+    pX = [p1.x; p2.x; p3.x; p4.x; p5.x; p6.x];
+    pY = [p1.y; p2.y; p3.y; p4.y; p5.y; p6.y];
+       
+    myfun = @(x,xdata)(exp(x(1))./ xdata.^exp(x(2))) - x(3);
+    xstart = [4, -2, 54];
+        
+    fittedmodel = lsqcurvefit(myfun,xstart,pX,pY, [], [], optimset('Display', 'off'));
+            
+    %FitError: mean squared error of the fit to the real spectrum in the band 8-15 Hz.
+    ts_8to15 = sp_ica.t(find(sp_ica.t == 8,1):find(sp_ica.t == 15,1)); 
+    fs_8to15 = sp_ica.x(find(sp_ica.t == 8,1):find(sp_ica.t == 15,1),ic)'; 
+    fiterror = log(norm(myfun(fittedmodel, ts_8to15)-fs_8to15)^2); 
+    
+    %lambda: parameter of the fit
+    lambda = fittedmodel(2); 
+    
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    % Averaged local skewness 15s
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    interval = 15; 
+    abs_local_scewness = [];
+    for i=1:interval:length(cnt.x(:,ic))/cnt.fs-interval
+        abs_local_scewness = [abs_local_scewness, abs(skewness(cnt.x(i*cnt.fs:(i+interval)*cnt.fs, ic)))];
+    end
+    mean_abs_local_scewness_15 = log(mean(abs_local_scewness)); 
+
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    % current density norm
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    norm_dipole_modelling = log(norm(M100*A(:,ic)));
+
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    %SpatialRange
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    spatial_range = log(max(A(:,ic)) - min(A(:,ic)));
+
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    %Append Features 
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+   %features(:,ic)= [norm_dipole_modelling, spatial_range,  mean_abs_local_scewness_15, lambda,Hz8_13, fiterror]; 
+   features(:,ic)= [norm_dipole_modelling, spatial_range,   ...
+       mean_abs_local_scewness_15, lambda, Hz8_13, fiterror]; 
+end 
+
+
+function [M100, idx_clab_desired] = get_M100(clab_desired)
+% [M100, idx_clab_desired] = get_M100(clab_desired)
+%
+% IN  clab_desired - channel setup for which M100 should be calculated
+% OUT M100 
+%     idx_clab_desired
+% M100 is the matrix such that  feature = norm(M100*ica_pattern(idx_clab_desired), 'fro')
+%
+% Stefan Haufe
+
+lambda = 100;
+
+load inv_matrix_icbm152.mat; %L (forward matrix 115 x 2124 x 3), clab (channel labels)

```

Comment:
try
    loadData = load(fullfile(BTB.Dir, 'external', 'mara', 'inv_matrix_icbm152.mat'))
    L = loadData.L
    clab = loadData.clab
catch
    error(sprintf('could not load dependencies of MARA. \nMake sure that you have have successfully downloaded the requested files with \n\n> bbci_import_dependencies(''MARA'') '))
end